### PR TITLE
update react-hyperscript

### PR DESCRIPTION
**Pull request by roshal**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test react-hyperscript`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [x] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [x] Delete the package's directory.
- [x] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by roshal** [comment]
@mlmorg fyi

File: types/react-hyperscript/index.d.ts [hunk 1]
```diff
@@ -1,28 +1,145 @@
 // Type definitions for react-hyperscript 3.0
 // Project: https://github.com/mlmorg/react-hyperscript
-// Definitions by: tock203 <https://github.com/tock203>
+// Definitions by: roshal <https://github.com/roshal>
```
Comment (orta) [inline types/react-hyperscript/index.d.ts:3]: Did @tock203 ask to be removed? you didn't mention in the PR body

Comment (roshal) [reply types/react-hyperscript/index.d.ts:3]: @tock203 no longer exists

**Comment by roshal** [comment]
Ready to merge

**Comment by roshal** [comment]
reopen this pull request please

**Comment by roshal** [comment]
@orta

**Comment by orta** [comment]
I don't know if the bot will handle re-opening a PR from last year, you might be better off re-creating it if this doesn't work out

**Comment by orta** [comment]
it has a bunch of conflicts - so I'd recommend re-creating otherwise

**Comment by roshal** [comment]
@orta rebase ready

**Review by orta — APPROVED** [review]
Cool, thanks

**Comment by roshal** [comment]
ready to merge

**Pull request merged**